submarine-server/server-core/src/main/java/org/apache/submarine/server/s3/Client.java

```diff
@@ -0,0 +1,175 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.submarine.server.s3;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import javax.ws.rs.core.Response;
+import io.minio.GetObjectArgs;
+import io.minio.ListObjectsArgs;
+import io.minio.MinioClient;
+import io.minio.PutObjectArgs;
+import io.minio.RemoveObjectsArgs;
+import io.minio.Result;
+import io.minio.messages.DeleteError;
+import io.minio.messages.DeleteObject;
+import io.minio.messages.Item;
+import org.apache.submarine.commons.utils.exception.SubmarineRuntimeException;
+
+
+public class Client {
+  public MinioClient minioClient;
+
+  public Client() {
+    minioClient = MinioClient.builder()
+        .endpoint(S3Constants.ENDPOINT)
+        .credentials(S3Constants.ACCESSKEY, S3Constants.SECRETKEY)
+        .build();
+  }
+
+  public Client(String endpoint) {
+    minioClient = MinioClient.builder()
+        .endpoint(endpoint)
+        .credentials(S3Constants.ACCESSKEY, S3Constants.SECRETKEY)
+        .build();
+  }
+
+  /**
+   * Get a list of artifact path under the experiment
+   *
+   * @param experimentId experiment id
+   * @return a list of artifact path
+   */
+  public List<String> listArtifactByExperimentId(String experimentId) throws SubmarineRuntimeException {
+    Iterable<Result<Item>> artifactNames = minioClient.listObjects(ListObjectsArgs.builder()
+        .bucket(S3Constants.BUCKET).prefix(experimentId + "/").delimiter("/").build());
+    List<String> response = new ArrayList<>();
+    Iterable<Result<Item>> artifacts;
+    for (Result<Item> artifactName : artifactNames) {
+      try {
+        artifacts = minioClient.listObjects(ListObjectsArgs.builder().bucket(S3Constants.BUCKET)
+            .prefix(artifactName.get().objectName()).delimiter("/").build());
+        for (Result<Item> artifact: artifacts) {
+          response.add("s3://" + S3Constants.BUCKET + "/" + artifact.get().objectName());
+        }
+      } catch (Exception e) {
+        throw new SubmarineRuntimeException(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode(),
+          e.getMessage());
+      }
+    }
+    return response;
+  }
+
+  /**
+   * Delete all the artifacts under given experiment name
+   *
+   * @param experimentId experiment id
+   */
+  public void deleteArtifactsByExperiment(String experimentId) {
+    deleteAllArtifactsByFolder(experimentId);
+  }
+
+  /**
+   * Delete all the artifacts under s3://submarine
+   */
+  public void deleteAllArtifacts() {
+    deleteAllArtifactsByFolder("");
+  }
+
+  /**
+   * Download an artifact
+   *
+   * @param path artifact path
+   * @return an array of byte
+   */
+  public byte[] downloadArtifact(String path) {
+    InputStream is;
+    byte[] buffer;
+    int b;
+    try {
```

Comment:
It's better to use the [try-with-resources Statement
](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Reply:
ok